tests/main/kernel-snap-refresh-on-core/task.yaml

```diff
@@ -0,0 +1,99 @@
+summary: Check that the kernel snap can be refreshed on a core device
+
+systems: [ubuntu-core-16-64]
+
+details: |
+    This test checks that the kernel snap can be refreshed from an installed
+    revision to a new one. It expects to find a new snap revision in the
+    channel pointed by the NEW_CORE_CHANNEL env var.
+
+manual: true
+
+environment:
+    KERNEL_CHANNEL: stable
+    NEW_KERNEL_CHANNEL: edge
+
+restore: |
+    rm -f prevBoot nextBoot
+        
+prepare: |
+    snap install test-snapd-tools
+
+execute: |
+    wait_core_post_boot() {
```

Comment:
Is it worth moving to boot.sh? Any chance it's going to be useful for other tests?

Reply:
A good question, I have a look at the other boot related tests